async_operation.go

```diff
@@ -0,0 +1,260 @@
+package swag
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"go/ast"
+	"log"
+	"regexp"
+	"strings"
+
+	typeSpec "github.com/go-openapi/spec"
+	"github.com/swaggest/go-asyncapi/spec-2.4.0"
+)
+
+type OperationAction string
+type Attribute string
+
+const (
+	Send    OperationAction = "send"
+	Receive OperationAction = "receive"
+)
+
+const (
+	serverAttr Attribute = "@server"
+	channelAttr Attribute = "@channel"
+	operationAttr Attribute = "@operation"
+)
+
+const (
+	openAPISchemaPrefix = "#/definitions/"
+	asyncAPISchemaPrefix = "#/components/schemas/"
+)
+
+type AsyncScope struct {
+	parser *Parser
+	servers map[string]*spec.ServersAdditionalProperties
+	channels map[string]*spec.ChannelItem
+	operations map[string]*OperationWithChannel
+}
+
+type OperationWithChannel struct {
+	action OperationAction
+	channel string
+	spec.Operation
+}
+
+// NewAsyncOperation creates a new AsyncOperation with default properties.
+func NewAsyncScope(parser *Parser) *AsyncScope {
+	if parser == nil {
+		parser = New()
+	}
+
+	asyncOperation := &AsyncScope{
+		parser:           parser,
+		servers:          make(map[string]*spec.ServersAdditionalProperties),
+		channels:         make(map[string]*spec.ChannelItem),
+		operations:       make(map[string]*OperationWithChannel),
+	}
+
+	return asyncOperation
+}
+
+// AttributeHandler is a map of attribute to the function that handles the attribute.
+var AttributeHandler = map[Attribute]func(*AsyncScope, *string, string, *ast.File) error {
+	serverAttr:  (*AsyncScope).ParseServerComment,
+	channelAttr: (*AsyncScope).ParseChannelComment,
+	operationAttr: (*AsyncScope).ParseOperationComment,
+}
+
+// ParseAsyncAPIComment parses the comment line and sets the AsyncAPI properties.
+func (asyncScope *AsyncScope) ParseAsyncAPIComment(funcName *string, comment string, astFile *ast.File) error {
+	commentLine := strings.TrimSpace(strings.TrimLeft(comment, "/"))
+	if len(commentLine) == 0 {
+		return nil
+	}
+
+	fields := FieldsByAnySpace(commentLine, 2)
+	attribute := fields[0]
+	lowerAttribute := strings.ToLower(attribute)
+	
+	var lineRemainder string
+	if len(fields) > 1 {
+		lineRemainder = fields[1]
+	}
+
+	handler, exists := AttributeHandler[Attribute(lowerAttribute)]
+	if exists {
+		return handler(asyncScope, funcName, lineRemainder, astFile)
+	}
+	
+	return fmt.Errorf("unknown attribute '%s' in comment '%s'", attribute, comment)
```

Comment:
Keep the happy path closer to the left for ease of read.

```suggestion
	if !exists {
		return fmt.Errorf("unknown attribute '%s' in comment '%s'", attribute, comment)
	}
	return handler(asyncScope, funcName, lineRemainder, astFile)
```

Reply:
donee